File: bookmysalon-backend/models/payment.go
// bookmysalon/models/payment.go

package models

// Transaction represents a payment transaction in the system.
// swagger:model
type Transaction struct {
	// The unique ID for the transaction.
	//
	// required: true
	// example: 1001
	TransactionID int `json:"transaction_id"`

	// The ID of the user who initiated the transaction.
	//
	// required: true
	// example: 1
	UserID int `json:"user_id"`

	// The amount involved in the transaction.
	//
	// required: true
	// example: 59.99
	Amount float64 `json:"amount"`

	// The date of the transaction.
	//
	// required: true
	// example: "2023-05-20"
	Date string `json:"date"`

	// The status of the transaction.
	//
	// required: true
	// example: "Completed"
	Status string `json:"status"`

	// The method of payment used.
	//
	// required: true
	// example: "Credit Card"
	PaymentMethod string `json:"payment_method"`
}

// Invoice represents an invoice linked to a transaction in the system.
// swagger:model
type Invoice struct {
	// The unique ID for the invoice.
	//
	// required: true
	// example: 2001
	InvoiceID int `json:"invoice_id"`

	// The transaction ID associated with the invoice.
	//
	// required: true
	// example: 1001
	TransactionID int `json:"transaction_id"`

	// The details or items on the invoice.
	//
	// required: true
	// example: "Haircut, Styling"
	Details string `json:"details"`

	// The date when the invoice was issued.
	//
	// required: true
	// example: "2023-05-21"
	DateIssued string `json:"date_issued"`
}

// Promotion represents a promotional offer or discount in the system.
// swagger:model
type Promotion struct {
	// The unique ID for the promotion.
	//
	// required: true
	// example: 3001
	PromotionID int `json:"promotion_id"`

	// A description of the promotion.
	//
	// required: true
	// example: "Summer Special Discount"
	Description string `json:"description"`

	// The amount of discount provided by the promotion.
	//
	// required: true
	// example: 10.00
	DiscountAmount float64 `json:"discount_amount"`

	// The start date for the promotion's validity.
	//
	// required: true
	// example: "2023-06-01"
	ValidFrom string `json:"valid_from"`

	// The end date for the promotion's validity.
	//
	// required: true
	// example: "2023-08-31"
	ValidTo string `json:"valid_to"`
}

File: bookmysalon-app/lib/models/payment.dart
]// Transaction Model
class Transaction {
  final int transactionID;
  final int userID;
  final double amount;
  final String date;
  final String status;
  final String paymentMethod;

  Transaction({
    this.transactionID,
    this.userID,
    this.amount,
    this.date,
    this.status,
    this.paymentMethod,
  });

  factory Transaction.fromJson(Map<String, dynamic> json) {
    return Transaction(
      transactionID: json['transaction_id'],
      userID: json['user_id'],
      amount: json['amount'].toDouble(),
      date: json['date'],
      status: json['status'],
      paymentMethod: json['payment_method'],
    );
  }
}

// Invoice Model
class Invoice {
  final int invoiceID;
  final int transactionID;
  final String details;
  final String dateIssued;

  Invoice({
    this.invoiceID,
    this.transactionID,
    this.details,
    this.dateIssued,
  });

  factory Invoice.fromJson(Map<String, dynamic> json) {
    return Invoice(
      invoiceID: json['invoice_id'],
      transactionID: json['transaction_id'],
      details: json['details'],
      dateIssued: json['date_issued'],
    );
  }
}

// Promotion Model
class Promotion {
  final int promotionID;
  final String description;
  final double discountAmount;
  final String validFrom;
  final String validTo;

  Promotion({
    this.promotionID,
    this.description,
    this.discountAmount,
    this.validFrom,
    this.validTo,
  });

  factory Promotion.fromJson(Map<String, dynamic> json) {
    return Promotion(
      promotionID: json['promotion_id'],
      description: json['description'],
      discountAmount: json['discount_amount'].toDouble(),
      validFrom: json['valid_from'],
      validTo: json['valid_to'],
    );
  }
}
